Don't assume user making the PR is hosting the repository (or that it's named "ocaml")

The generated PR currently makes a couple of assumptions which don't always hold, namely that the user who made the PR is the user hosting the repository and that the repository is named "ocaml" ([see here](https://github.com/avsm/opam-sync-github-prs/blob/add626d65afe0e87cb68981da63adb9470e313cd/generate.ml#L38)).

Here's an example of things going wrong:

``` bash
$ cat compilers/4.03.0/4.03.0+pr132/4.03.0+pr132.comp 
opam-version: "1"
version: "4.03.0"
src: "https://github.com/lefessan/ocaml/archive/flambda_trunk.tar.gz"
build: [
  ["./configure" "-prefix" prefix "-with-debug-runtime"]
  [make "world"]
  [make "world.opt"]
  [make "install"]
]
packages: [ "base-unix" "base-bigarray" "base-threads" ]
env: [[CAML_LD_LIBRARY_PATH = "%{lib}%/stublibs"]]
$ 
opam switch 4.03.0+pr132
$ [compiler.get] Downloading https://github.com/lefessan/ocaml/archive/flambda_trunk.tar.gz
[ERROR] curl: code 404 while downloading https://github.com/lefessan/ocaml/archive/flambda_trunk.tar.gz
[ERROR] https://github.com/lefessan/ocaml/archive/flambda_trunk.tar.gz is not available.
```

--- a/generate.ml
+++ b/generate.ml
@@ -26,6 +26,14 @@ let make_compiler_spec ~version ~output_dir pull =
   let subdir = sprintf "%s/compilers/%s/%s/" output_dir version name in
   let source_user = pull.pull_user.user_login in
   let head = pull.pull_head.branch_ref in
+  let source_url =
+    match pull.pull_head.branch_repo with
+      Some repo -> 
+        sprintf "https://github.com/%s/archive/%s.tar.gz"
+         repo.repo_full_name head
+    | None ->
+         sprintf "https://github.com/%s/ocaml/archive/%s.tar.gz" source_user head
+  in
   printf "Generating: %s\n%!" subdir;
   Unix.mkdir_p subdir;
   let open Out_channel in
@@ -35,7 +43,7 @@ let make_compiler_spec ~version ~output_dir pull =
         [
           "opam-version: \"1\"";
           sprintf "version: \"%s\"" version;
-          sprintf "src: \"https://github.com/%s/ocaml/archive/%s.tar.gz\"" source_user head;
+          sprintf "src: \"%s\"" source_url;
           "build: [";
           "  [\"./configure\" \"-prefix\" prefix \"-with-debug-runtime\"]";
           "  [make \"world\"]";

